app/models/system.rb

```diff
@@ -76,6 +80,35 @@ def update_instance_data(instance_data)
     update!(instance_data: instance_data)
   end
 
+  def complete_profiles=(profiles_hash)
```

Comment:
Is there a benefit of this method over just using `system.profiles = provided_profiles`?

Reply:
It means that we can just include a complete_profilee in the parameters block passed to a create or update operations, which simplifies the integration with the existing code for the associated handlers.

It originated from a suggestion from Genimi on how best to integrate the proposed conceptual changes into the existing RMT code base with minimal changes to existing code.

If you prefer me to make it more explicit I can change it to do so.